src/diskquota.h

```diff
@@ -75,6 +75,14 @@ extern int diskquota_worker_timeout;
 #define DiskquotaGetRelstorage(classForm) (0)
 #endif /* GP_VERSION_NUM */
 
+typedef struct
+{
+	bool is_connected;
+	bool is_active_snapshot_pushed;
+	bool do_commit;
+	bool is_under_transaction;
+} SPI_state;
```

Comment:
What about defining SPI_state as unsigned char and using flags like
```
enum
{
	is_connected = 1 << 0,
	is_active_snapshot_pushed = 1 << 1,
	do_commit  = 1 << 2,
	is_under_transaction   = 1 << 3,
};
```
?

In this case we can avoid using pointers

Reply:
applied